test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Evaluation/AutoComputePayloadMetadataInJsonIntegrationTests.cs

```diff
@@ -1356,6 +1375,172 @@ private static IEdmModel GetModelWithAbstractType(out EdmEntitySet entitySet)
             return model;
         }
 
+        [Theory]
+        [InlineData("minimal")]
+        [InlineData("full")]
+        public void WritingContainedNavigationPropertyWithoutProvideKeyPropertyOnParentResourceWorks(string metadataLevel)
+        {
+            MemoryStream outputStream = new MemoryStream();
+            var container = ContainerBuilderHelper.BuildContainer(null);
+            IODataResponseMessage message = new InMemoryMessage() { Stream = outputStream, Container = container };
+
+            message.SetHeader("Content-Type", metadataLevel == "full" ? "application/json;odata.metadata=full" : "application/json");
+            ODataMessageWriterSettings settings = new ODataMessageWriterSettings();

```

Comment:
>      ODataMessageWriterSettings settings = new ODataMessageWriterSettings(); [](http://example.com/codeflow?start=6&length=77)

for better readability, move this down to immediately before it's used in 1401?

Reply:
#Resolved.